# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../loudmouth/loudmouth_gnutls.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
diff -r -u -N loudmouth-0.17.2.orig/loudmouth/lm-ssl.c loudmouth-0.17.2/loudmouth/lm-ssl.c
--- loudmouth-0.17.2.orig/loudmouth/lm-ssl.c	2004-10-19 09:52:48.000000000 +0200
+++ loudmouth-0.17.2/loudmouth/lm-ssl.c	2005-03-28 02:41:04.520131624 +0200
@@ -71,7 +71,6 @@
 	}
 	
 	if (status & GNUTLS_CERT_INVALID
-	    || status & GNUTLS_CERT_NOT_TRUSTED
 	    || status & GNUTLS_CERT_REVOKED) {
 		if (ssl->func (ssl, LM_SSL_STATUS_UNTRUSTED_CERT,
 			       ssl->func_data) != LM_SSL_RESPONSE_CONTINUE) {
@@ -126,7 +125,7 @@
 
 		gnutls_x509_crt_deinit (cert);
 
-		if (gnutls_x509_fingerprint (GNUTLS_DIG_MD5, &cert_list[0],
+		if (gnutls_fingerprint (GNUTLS_DIG_MD5, &cert_list[0],
 					     ssl->fingerprint,
 					     &digest_size) >= 0) {
 			if (ssl->expected_fingerprint &&
